lib/ecto/adapters/postgres/connection.ex

```diff
@@ -380,6 +381,22 @@ if Code.ensure_loaded?(Postgrex) do
        end)]
     end
 
+    defp window(%Query{windows: []}, _sources), do: []
+    defp window(%Query{windows: windows} = query, sources) do
+      [" WINDOW " |
+       intersperse_map(windows, ", ", fn
+         {name, definition} ->
+           [Atom.to_string(name), " AS ", partition_by(definition, sources, query)] end)]
```

Comment:
We need to use `quoted_name` or similar instead of `Atom.to_string(name)` to avoid SQL injection attacks.